src/app/shared/dovetail/dovetail-audio.ts

```diff
@@ -251,24 +235,22 @@ export class DovetailAudio extends ExtendableAudio {
     return this.currentAdzerkPromise;
   }
 
-  private calculateArrangement(
-    arrangementTemplate: DovetailArrangementEntry[], response: AdzerkResponse) {
-    let result: DovetailArrangementEntry[] = [];
-    for (let entry of arrangementTemplate) {
-      if (response.decisions[entry.id]) {
-        result.push(entry);
-        entry.audioUrl = response.decisions[entry.id].contents[0].data.imageUrl;
-        entry.duration = 10;
-        entry.impressionUrl = response.decisions[entry.id].impressionUrl;
+  private calculateSegments(dovetail: DovetailResponse, adzerk: AdzerkResponse) {
+    let result: DovetailSegment[] = [], isFirst = true;
+    for (let entry of dovetail.arrangement) {
+      if (adzerk.decisions[entry.id]) {
+        result.push(new DovetailSegment(entry, adzerk.decisions[entry.id], dovetail.tracker, isFirst));
```

Comment:
We have to track which segment is first, since _only_ it should log the download.